src/background.js

```diff
@@ -1,5 +1,209 @@
-browser.browserAction.onClicked.addListener(async () => {
-  await browser.tabs.create({
-    url: "debug-page/index.html",
-  });
+const configByAddonId = {
+  "messaging-system-personalization-experiment-1-addon-control@mozilla.org": {
+    branch: "control",
+    asRouterCfrProviderConfig: {
+      bucket: "cfr-control",
+      cohort: "PERSONALIZATION_EXPERIMENT_1_CONTROL",
+    },
+  },
+  "messaging-system-personalization-experiment-1-addon-treatment@mozilla.org": {
+    branch: "treatment",
+    asRouterCfrProviderConfig: {
+      bucket: "cfr-experiment",
+      cohort: "PERSONALIZATION_EXPERIMENT_1_TREATMENT",
+    },
+    scoreThreshold: 5000,
+  },
+};
+const config = configByAddonId[browser.runtime.id];
+const { bucket, cohort } = config.asRouterCfrProviderConfig;
+console.info("Experiment add-on background.js", browser.runtime.id);
+console.debug({ config });
+const periodicAlarmName = `${browser.runtime.id}:periodicCfrModelsPolling`;
+
+const onError = e => {
+  console.error(e);
+  browser.normandyAddonStudy.endStudy("CAUGHT_ERROR");
+};
+
+const run = async () => {
+  try {
+    // TODO - double-check that study has not already ended
+    const study = await browser.normandyAddonStudy.getStudy();
+    console.debug({ study });
+
+    const { hasRunInitiationAtLeastOnce } = await browser.storage.local.get(
+      "hasRunInitiationAtLeastOnce",
+    );
+    if (hasRunInitiationAtLeastOnce !== true) {
+      await firstRun();
+      await browser.storage.local.set({ hasRunInitiationAtLeastOnce: true });
+    }
+    await everyRun();
+  } catch (e) {
+    onError(e);
+  }
+};
+
+const firstRun = async () => {
+  console.info(`Force syncing "${bucket}" bucket contents`);
+  await browser.privileged.remoteSettings.clearLocalDataAndForceSync(bucket);
+
+  if (config.branch === "control") {
+    console.info("Writing hard-coded provider cfr pref", { bucket, cohort });
+    await browser.privileged.messagingSystem.setASRouterCfrProviderPref(
+      bucket,
+      cohort,
+    );
+    return;
+  }
+
+  console.info("Writing hard-coded score threshold", config.scoreThreshold);
+  await browser.privileged.personalizedCfrPrefs.setScoreThreshold(
+    config.scoreThreshold,
+  );
+};
+
+const everyRun = async () => {
+  try {
+    // Do nothing on every run in control branch
+    if (config.branch === "control") {
```

Comment:
I wonder if this could be a constant somewhere just do avoid possible misconfiguration?

Reply:
Not a constant currently, but definitely [hard-coded](https://github.com/mozilla/messaging-system-personalization-experiment-1-addon/pull/2/files/d08f5e52b66009a8c39dfa0fe0bac6952c158b5d#diff-961641dd6af66f5e3313307cf4eb8f1dR18), so I think the risk for misconfiguration is minimal - the behavior that gets tested in QA also gets shipped.